### PR TITLE
[JENKINS-27081] Executors are dying because SortedIntList wrongly returns nextBuildNumber is already in use.

### PR DESCRIPTION
**Pull request by brunocarneiro**
Executors start dying after upgrading Jenkins to 1.599 and having all jobs configured to discard old builds.
SortedIntList.isInRange was not considering the list content.

Check the problem:
https://issues.jenkins-ci.org/browse/JENKINS-27081

### CODE REVIEW

File: core/src/main/java/jenkins/model/lazy/SortedIntList.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ public int ceil(int v) {
     }
 
     public boolean isInRange(int idx) {
-        return 0<=idx && idx<size;
+        return find(idx)>-1;
```
Comment (olivergondza) [inline core/src/main/java/jenkins/model/lazy/SortedIntList.java]: This should be trivial to unittest.

**Comment by jtnord** [comment]
I don't believe the big report (as written) is a defect but rather correct behavior.

File: core/src/main/java/jenkins/model/lazy/SortedIntList.java [hunk 2]
```diff
@@ -131,7 +131,7 @@ public int ceil(int v) {
     }
 
     public boolean isInRange(int idx) {
-        return 0<=idx && idx<size;
+    	return 0<=idx && idx<size;
```
Comment (KostyaSha) [inline core/src/main/java/jenkins/model/lazy/SortedIntList.java]: exclude whitespace changes

Comment (brunocarneiro) [inline core/src/main/java/jenkins/model/lazy/SortedIntList.java]: Strange....
This line is indented correctly.
Look at: https://github.com/brunocarneiro/jenkins/blob/proposeNewNumberAlreadyInUse/core/src/main/java/jenkins/model/lazy/SortedIntList.java

Comment (daniel-beck) [inline core/src/main/java/jenkins/model/lazy/SortedIntList.java]: @brunocarneiro It's indented by four spaces and a tab.

**Comment by daniel-beck** [comment]
This does not appear to be correct. In fact, the referenced report indicates correct behavior by Jenkins core. If there's a build with number 5288 already, any build number lower than 5289 is invalid for a new build. The error message clearly states (emphasis mine):

> cannot create a build with number 1447 since that (**or higher**) is already in use among […]

 In fact, this seems to be a duplicate of [JENKINS-26739](https://issues.jenkins-ci.org/browse/JENKINS-26739) (which contains some more information about the problem related to Matrix Project).

**Pull request closed without merging**